utils/data.hpp

```diff
@@ -0,0 +1,143 @@
+    Qt::BDiagPattern, Qt::FDiagPattern, Qt::DiagCrossPattern,
+    Qt::LinearGradientPattern, Qt::RadialGradientPattern,
+    Qt::ConicalGradientPattern
+};
+const QStringList TEXT_ALIGN_STR = {
+    "AlignLeft", "AlignRight", "AlignTop",
+    "AlignBottom", "AlignCenter"
+};
+const QList<Qt::AlignmentFlag> TEXT_ALIGN_VAL = {
+  Qt::AlignLeft, Qt::AlignRight, Qt::AlignTop,
+    Qt::AlignBottom, Qt::AlignCenter
+};
+const QStringList TEXT_STYLE_STR = {
+  "StyleNormal", "StyleItalic", "StyleOblique"
+};
+const QList<QFont::Style> TEXT_STYLE_VAL = {
+  QFont::StyleNormal, QFont::StyleItalic, QFont::StyleOblique
+};
+const QStringList TEXT_WEIGHT_STR = {
+  "Thin", "ExtraLight", "Light", "Normal", "Medium",
+    "DemiBold", "Bold", "ExtraBold", "Black"
+};
+const QList<QFont::Weight> TEXT_WEIGHT_VAL = {
+  QFont::Thin, QFont::ExtraLight, QFont::Light, QFont::Normal, QFont::Medium,
+    QFont::DemiBold, QFont::Bold, QFont::ExtraBold, QFont::Black
+};
+
+class Data
+{
+private:
+    Canvas* mainCanvas; // Pointer to the main canvas of the program
+    QString currentDirectory;   // Current file loaded into the project
+
+    void GetTo(QTextStream&, QChar);  // Helper function moves the given text stream up to the given character
+    QPoint* GetPoints(QList<qint32>&);    // Helper function converts a list of integers to a pointer to a block of qpoitns
+
+    void ExtractFont(QTextStream&, QFont&);  // Extract data on the shape's font from the text stream specified
+
+    Shape* ConstructLine(QString&, QList<qint32>&/*, QPen*/);  // Construct a line-type using the given data
+    Shape* ConstructShape(QString&, QList<qint32>&/*, QPen, QBrush*/); // Construct a 2-d type shape with the given data
+    Shape* ConstructText(QString&, QList<qint32>&, QString&, QFont&); // Construct a text-type shape with the given data
+
+    // Given a parallel list of strings and enumerated types, try to
+    // match the string given to the list of enumerated types and return the enumerated type value
+    template<typename EnumType>
+    EnumType StringToEnum(const QString& identifier, const QStringList& identifierList, const QList<EnumType>& valueList)
+    {
+        try {
+            // Find the identifier in the list
+            int index = identifierList.indexOf(identifier);
+
+            // If the string was not found in the list, construct an error message and throw an exception
+            if(index < 0)
+            {
+                QString message;
+                QTextStream stream(&message);
+                stream << "Title " << identifier << " does not describe any enumerated type in the list";
+                GeneralException(message).raise();
+            }
+
+            return valueList[index];
+        } catch (GeneralException& exception) {
+            return valueList[0];
+        }
+    }
+
+public:
+    Data(Canvas*);  // Constructor initializes the main canvas
+
+    void Save();   // Output the shape info to the current directory
```

Comment:
Methods should be camelCase

Reply:
Ah right, I wasn't paying attention to the common style.  Good catch!